src/literal.h

```diff
@@ -732,6 +738,17 @@ struct GCData {
   GCData(HeapType type, Literals values) : type(type), values(values) {}
 };
 
+// The data of a (ref exn) literal.
+struct ExnData {
+  // The tag of this exn data.
+  Name tag;
+
+  // The payload of this exn data.
+  Literals payload;
+
+  ExnData(Name tag, Literals payload) : tag(tag), payload(payload) {}
+};
```

Comment:
If we identify exception tags by their name, how does that handle cross-module catches? I mean
```wat
(module $A
  (tag $foo (import "a" "b"))
  (func $throw (export "A.throw")
    (throw a $foo)
  )
)
(module $B
  (tag $bar (import "a" "b"))
  (func $catch
    (block $out
      (try_table (catch $bar $out)
        (call $A.throw)
      )
    )
  )
)
```
The same tag is imported into two different modules, but with different internal names. I think the catch here would not properly catch?

Reply:
Huh, that's a good point. It hadn't crossed my mind. I'll have to think about it tomorrow. I'll probably need some actual run-time representation of a tag, after all.

Reply:
I've investigated a bit, and I think this is a larger problem than what this PR tackles. It seems the existing support for `try` has the same issue: the thrown `WasmException::tag` is also the `Name` of the tag in one module, and if another module tries to catch the exception, the `Name`s might not match.

This seems to be an issue only for the `shell-interface.h`, which is the only one that actually implements `ExternalInterface::throwException` as throwing the `WasmException`.

AFAICT, the same issue also exists for `ref.func`s: a `Literal` of a function type holds a simple `Name`, and it does not seem like any effort is made to translate those `Name`s when they cross module boundaries. Later, a `call_ref` looks up that `Name` in the current module's function map:
https://github.com/WebAssembly/binaryen/blob/23a1a1aacc0dd14767ca8c53a034b7c6bb4acf52/src/wasm-interpreter.h#L4139

Reply:
Good points. RefFunc should have a reference to the actual function, not the name, and likewise for these others. Yeah, I agree this is a larger issue than this PR, and I guess we missed it because we don't currently test cross-module calls. Sounds good to ignore it here, but maybe add something like `TODO: handle cross-module calls using something other than a Name here`.